### PR TITLE
Add a paragraph component

### LINKED ISSUE #19: Paragraph component
- [ ] Basic Sketch components
- [x] React components
- [x] Documentation

### PR DESCRIPTION
**Pull request by sarahmonster**
Closes #19. 

This implements a baseline paragraph component + documentation. 

For future iterations, I'd like to be able to do the following:

- Change the bottom-margin (only)
- Add a drop cap styling (I do love a good drop cap)
- Change the colour (right now, it's just using default browser colour)

I'm wondering if maybe I should address the text colour issue now, rather than later. I'm thinking we should add a theme variable(s) for text colour—probably a few actually. Something like:

- TextColour, TextColourDark, TextColourLight
- TextColourInverse, TextColourInverseDark, TextColourInverseLight (for on dark backgrounds)

That would probably give us enough to play with, and they'd all be various shades of gray from the palette, but you could also pass whatever colour you wanted over.

### CODE REVIEW

**Comment by sarahmonster** [comment]
Oh, and this is stacked on top of #28, for obvious reasons.

**Comment by tofumatt** [comment]
> Oh, and this is stacked on top of #28, for obvious reasons.

Ouch, enjoy your merge conflicts then; I've already updated #28. 😆😉

**Comment by sarahmonster** [comment]
@tofumatt review me, please! I'm adding spacing variables in my next PR (....already in progress) and I'll do the drop-cap functionality in a separate PR, but I did add text colour options in this PR.

**Comment by sarahmonster** [comment]
Sidenote: accessibility guidelines regarding the amount of space between paragraphs are REALLY intense. Just saying.

File: src/components/Paragraph/README.md [hunk 1]
```diff
@@ -1,61 +1,105 @@
-## Available Sizes
-
-### Large Paragraph
+A `Paragraph` is a self-contained unit of one or more coherent sentences related to a single topic or idea.
 
 ```jsx
 <Paragraph size="large">
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: Looks like this ought to be:

```suggestion
<Paragraph large>
```

File: src/components/Paragraph/README.md [hunk 2]
```diff
@@ -1,61 +1,105 @@
...
+<Paragraph>
+	Vestibulum ornare eget lacus cursus rhoncus. Cras nec
+	lectus in sem viverra volutpat.
 </Paragraph>
 ```
 
-### Medium Paragraph
+## Usage
+
+Use a `Paragraph` for any body copy comprising of one or more full sentences.
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: I wonder if we'll want to write components like `<Paragraph>` to make it clear they're components? 🤷‍♂

File: src/components/Paragraph/README.md [hunk 3]
```diff
@@ -1,61 +1,105 @@
...
+
+Don't use a `Paragraph` for:
+- A single text link or call to action—use a [Button](../#/Function/Components/Button) instead.
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: Can we highlight "call to action" or separate it a bit better? I feel like this is all a bit tough to parse really quickly. I love me some dashes too but separate sentences might do better here. 🤷🏻‍♂️

```suggestion
- A single text link or _call to action_—use a [Button](../#/Function/Components/Button) instead.
```

File: src/components/Paragraph/README.md [hunk 4]
```diff
@@ -1,61 +1,105 @@
...
+- A heading or subheading—use a `Heading`, `Subtitle`, or `Page Title`.
+- A list of related, equally weighted items—use a `List` instead.
+
+## Appearance
+
+- Line lengths should be between 55-70 characters.
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: Some of these items have full-stops at the end and others don't. I think it's best to just be consistent and always use them in lists.

File: src/components/Paragraph/README.md [hunk 5]
```diff
@@ -1,61 +1,105 @@
...
+- Text colour should contrast with the background.
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: You should mention the minimum acceptable contrast here if you have it handy ("WCAG AAA", etc.)

File: src/components/Paragraph/README.md [hunk 6]
```diff
@@ -1,61 +1,105 @@
...
-	pharetra.
+	lectus in sem viverra volutpat.
 </Paragraph>
 ```
 
-## Colors
+Use an **inverse paragraph** if you need to put a paragraph on top of a dark background.
+
+```jsx
+<Paragraph inverse>
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: `inverted` is a better prop name, I feel like. Thoughts?

File: src/components/Paragraph/README.md [hunk 7]
```diff
@@ -1,61 +1,105 @@
...
+	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut bibendum hendrerit
+	augue gravida hendrerit. Vestibulum ornare eget lacus cursus rhoncus. Cras nec
+	lectus in sem viverra volutpat.
+</Paragraph>
+```
 
-### Paragraph with custom color
+For both inverse and default type colours, you can also use a dark and light variant. Use sparingly when needed to provide an additional level of hierarchy or emphasis.
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]:
```suggestion
For both `inverse` and default type colours, you can also use a dark and light variant. Use sparingly to provide an additional level of hierarchy or emphasis.
```

File: src/components/Paragraph/README.md [hunk 8]
```diff
@@ -1,61 +1,105 @@
...
 
 ```jsx
-<Paragraph color="accent">Aenean auctor justo nisl, in ullamcorper turpis pharetra mattis. Duis eget vulputate est. Nullam a velit pharetra, hendrerit magna vitae, accumsan magna.</Paragraph>
-<Paragraph color="medium">Aenean auctor justo nisl, in ullamcorper turpis pharetra mattis. Duis eget vulputate est. Nullam a velit pharetra, hendrerit magna vitae, accumsan magna.</Paragraph>
-<Paragraph color="light">Aenean auctor justo nisl, in ullamcorper turpis pharetra mattis. Duis eget vulputate est. Nullam a velit pharetra, hendrerit magna vitae, accumsan magna.</Paragraph>
-<Paragraph color="danger">Aenean auctor justo nisl, in ullamcorper turpis pharetra mattis. Duis eget vulputate est. Nullam a velit pharetra, hendrerit magna vitae, accumsan magna.</Paragraph>
-<Paragraph color="success">Aenean auctor justo nisl, in ullamcorper turpis pharetra mattis. Duis eget vulputate est. Nullam a velit pharetra, hendrerit magna vitae, accumsan magna.</Paragraph>
+<Paragraph light>
+	Light
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: Some mad tabs going on in here. You using Atom? 😆

File: src/components/Paragraph/README.md (hunk 5 above)
Comment (tofumatt) [reply src/components/Paragraph/README.md]: Reading 'til the end: just a link to the `#accessibility` header should do. 😄

File: src/components/Paragraph/README.md [hunk 9]
```diff
@@ -1,61 +1,105 @@
...
+When writing long passages of text comprised of many paragraphs, highlight important information using **strong** and _emphasis_ tags. Use headings and lists to make content easier to scan.
+
+## Accessibility
+
+- Ensure text colour has a contrast ratio of at least 7:1 (WCAG AAA).
+- Use relative units to ensure that text can be resized by users.
+- Line height should be at least a space-and-a-half within paragraphs. (1.5)
+- Paragraph spacing should be at least 1.5 times the line height.
+- Don't allow line lengths to span more than 80 characters.
+- Font size should be a minimum of 16px.
```
Comment (tofumatt) [inline src/components/Paragraph/README.md]: Even for `small` paragraphs? Might be worth clarifying (I feel like `14px` "small" paragraphs are okay).

File: src/components/Paragraph/index.js [hunk 10]
```diff
@@ -3,14 +3,64 @@ import PropTypes from 'prop-types';
...
 
 import { useTheme } from 'themes';
+import { bodySmall, bodyMedium, bodyLarge } from 'themes/mixins';
 
-export const Paragraph = ({ children, ...otherProps }) => {
+export const Paragraph = ({
+  children,
+  large,
+  small,
+  inverse,
```
Comment (tofumatt) [inline src/components/Paragraph/index.js]: As mentioned, I think I'd prefer:
```suggestion
  inverted,
```
for this prop name.

File: src/components/Paragraph/index.js [hunk 11]
```diff
@@ -3,14 +3,64 @@ import PropTypes from 'prop-types';
...
+          css`
+            ${css(bodyLarge(theme))};
+          `}
+
+        ${small &&
+          css`
+            ${css(bodySmall(theme))};
+          `}
+
+          ${light &&
```
Comment (tofumatt) [inline src/components/Paragraph/index.js]: These are indented a bit farther than I'd expect, eh? I guess I need to improve the style linting/prettier settings.

File: src/themes/nautilus/index.js [hunk 12]
```diff
@@ -99,38 +99,46 @@ const fonts = {
...
       warning: colors.yellow600,
       warningDark: colors.yellow800,
       warningLight: colors.yellow200,
-
       danger: colors.red600,
       dangerDark: colors.red800,
       dangerLight: colors.red200,
     },
+
+    text: {
```
Comment (tofumatt) [inline src/themes/nautilus/index.js]: Very nice 👍

File: src/components/Paragraph/README.md (hunk 2 above)
Comment (sarahmonster) [reply src/components/Paragraph/README.md]: We could! I'm not in love with the component names in backticks, so lemme try that.

File: src/components/Paragraph/README.md (hunk 3 above)
Comment (sarahmonster) [reply src/components/Paragraph/README.md]: Fun fact: my "component template" that I'm working on actually sets these up in separate sentences, so I think I agree.

File: src/components/Paragraph/README.md (hunk 4 above)
Comment (sarahmonster) [reply src/components/Paragraph/README.md]: Agreed! But I think all of the list items here do?

File: src/components/Paragraph/README.md (hunk 2 above)
Comment (tofumatt) [reply src/components/Paragraph/README.md]: You'd likely still want to include the backticks to highlight that they're code, so it'd be:

```md
Use a `<Paragraph>` for any body copy comprising of one or more full sentences.
```

instead of:

```md
Use a `Paragraph` for any body copy comprising of one or more full sentences.
```

File: src/components/Paragraph/README.md (hunk 4 above)
Comment (tofumatt) [reply src/components/Paragraph/README.md]: Maybe; I think it was just a general trend I was seeing in the docs.

File: src/components/Paragraph/README.md (hunk 6 above)
Comment (sarahmonster) [reply src/components/Paragraph/README.md]: "Inverse" means opposite; "inverted" means upside-down!

File: src/components/Paragraph/index.js (hunk 11 above)
Comment (sarahmonster) [reply src/components/Paragraph/index.js]: Yeah, the indentation on these is super weird and hard to follow. Happy to see a fix!

File: src/components/Paragraph/README.md (hunk 8 above)
Comment (sarahmonster) [reply src/components/Paragraph/README.md]: SURE AM. (_!_)

File: src/components/Paragraph/README.md (hunk 9 above)
Comment (sarahmonster) [reply src/components/Paragraph/README.md]: The default theme puts the small paragraphs at 16px, but I'm okay dialling this back a bit since I think the spec only says that type needs to be resizeable up to 200%.
